bin/main.ml

```diff
@@ -1,18 +1,55 @@
 open Diffcessible
 open Cmdliner
+open ExtUnix.Specific
+
+let create_empty_diff_content () =
+  "diff --git a/Error b/Error\n--- a/Error\n+++ b/Error\n"
+
+let setup_input = function
+  | Some path ->
+      if not (Sys.file_exists path) then failwith ("File not found: " ^ path);
+      let ic = In_channel.open_bin path in
+      (Some ic, Notty_unix.Term.create (), None)
+  | None ->
+      let tty_path = ttyname Unix.stdout in
+      let tty_fd = Unix.openfile tty_path [ Unix.O_RDWR ] 0o600 in
+      let term = Notty_unix.Term.create ~input:tty_fd ~output:tty_fd () in
+      if not (Unix.isatty Unix.stdin) then (Some In_channel.stdin, term, None)
+      else (None, term, Some tty_fd)
+
+let safe_read_input ic is_stdin =
+  try
+    let content = In_channel.input_all ic in
+    if not is_stdin then In_channel.close ic;
+    content
+  with _ ->
+    if not is_stdin then In_channel.close ic;
+    create_empty_diff_content ()
 
 let main file_path =
-  let file = In_channel.open_bin file_path in
-  let s = In_channel.input_all file in
-  let patch = Patch.to_diffs s in
-  Interactive_viewer.start patch
+  let ic_opt, term, tty_fd_opt = setup_input file_path in
+  let input_content =
+    match ic_opt with
+    | None -> create_empty_diff_content ()
+    | Some ic -> safe_read_input ic (ic == In_channel.stdin)
+  in
+  let patch = Patch.to_diffs input_content in
+  Interactive_viewer.start ~term patch;
+  Notty_unix.Term.release term;
+  match tty_fd_opt with Some tty_fd -> Unix.close tty_fd | None -> ()
```

Comment:
@Julow What do you think of this strategy to handle closing tty_fd?

Reply:
That's the right time to close the FD but it doesn't look very safe. The FD wouldn't be closed in case of an exception.